api/src/main/java/org/apache/iceberg/io/ParallelIterable.java

```diff
@@ -28,25 +28,39 @@
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Future;
 import org.apache.iceberg.exceptions.RuntimeIOException;
-import org.apache.iceberg.io.CloseableGroup;
-import org.apache.iceberg.io.CloseableIterable;
-import org.apache.iceberg.io.CloseableIterator;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.relocated.com.google.common.collect.Iterables;
 
+/**
+ * Run iterables in parallel.
+ * @deprecated please use {@link CloseableIterable#combine(Iterable, ExecutorService, int)} instead.
+ */
+@Deprecated
```

Comment:
This class is not deprecated. It is still public and will not be removed. This should be created using `CloseableIterable#combine` rather than directly using the constructor. Can you remove this deprecation?

Reply:
I see the module here also changed. I don't think we are allowed to move public class to another module?

Reply:
Good catch, @jackye1995!